# team_Methylhomies


Team
------------

**Table of group members with their background, degree, affiliations and job assignments.**

| Name | Degree/Affiliation | Speacilizations |Job Assignments | 
| ------------- | ------------- | ------------- | ------------- |
| Samantha Schaffner | Medical Genetics  | Neuroepigenetics of human development and Parkinson’s Disease | DMR analysis and help with data processing and normalization |
| Cassia Warren | Interdisciplinary Oncology   | Biomarker development for Vitamin C treatment for Pancreatic cancer | Maintain/organize Github and Cell type correction analysis|
| Hilary Brewis  | Medical Genetics  | Histone variants and chromatin structure | Normalization and data processing |
| Randip Gill  | Educational Psychology  | Social epigenetics | DMR analysis |
| Lisa Wei | Bioinformatics  | Cancer genomics, sequence analysis | Help maintain/organize Github and Overall methylation level differences between brain subregions |

Our project uses data that was generated for the paper [Interindividual methylomic variation across blood, cortex, and cerebellum: implications for epigenetic studies of neurological and neuropsychiatric phenotypes. Hannon *et al* 2015](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/background_information/Interindividual%20methylomic%20variation%20across%20blood%20cortex%20and%20cerebellum%20implications%20for%20epigenetic%20studies%20of%20neurological%20and%20neuropsychiatric.pdf)

An outlne of our project can be found in our [Project proposal](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/project_proposal.md)

Introduction
------------

DNA methylation (DNAm), or covalent attachment of a methyl group to the 5’ position of cytosine bases located in CpG dinucleotides, plays a crucial role in maintaining patterns of gene expression during human development and aging. Along with other epigenetic modifications, DNAm is sensitive to environmental influences and can change over an individual’s lifespan. Thus, understanding the human methylome is important for determining both biomarkers for - and direct pathways implicating - health and disease [1]. Aberrant DNAm patterns have been correlated with common neurodegenerative disorders, including Alzheimer’s Disease and Parkinson’s Disease [2], as well as with mental disorders such as schizophrenia [3,4]. However, much of the literature on either the diseased or healthy brain methylome fails to separate DNAm data by cell type composition - a major driver of DNAm variability - or by brain region [5-8]. 

Motivation
-----------

Objectives
----------


Data
-----

Analysis and tasks directory
----------------------------

A summary of major analysis, their motivations, and their main results
----------------------------------------------------------------------

Discussion
------------


Deliverables
--------------

- [Project proposal](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/project_proposal.md)
- [Progress report](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/progress_report.md)
- [Poster](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/Poster.pdf)

Repo Directory
----------

We aim to study the differences in methylation between cerebellum and cortex. For more information on our project please visit our project proposal [here](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/project_proposal.md)

We will be using methylation data from Hannon *et al.* paper that is publically available. More information about our data set please check it out [here](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/data)

Our [data folder](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/data) consists of two subfolders
  - [raw data](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/data/raw_data)
  - [processed data](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/data/processed_data)
  our data set is too large to upload into the repo but we have included the raw and processed meta data.
  
our [background information folder](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/background_information) is there to provide your with some relevant papers to our project

our [src folder](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/src) is where the codes used for our analysis are stored. 
  - in [working codes](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/src/working_codes) you will find parts of codes that were created and then shared to help create code for our final analysis codes. Explain through the [README](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/src/working_codes/README.md)
  - in [final codes](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/src/final_codes) We have the final copies of our code used to optain our results. Detailed explanation of which step they were used for can be found in the [README](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/src/final_codes/README.md)
  
our [results folder](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/results) contains the decriptions of the aims and conclusions of our analysis steps and the important final images derived from the analysis

other folders

our [resources folder](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/resources) includes codes that we modified for our analysis. These codes have been referenced in our src where appropriate

our [images folder](https://github.com/STAT540-UBC/team_Methylhomies/tree/master/Images) includes images we shared with eachother to keep updates on the project and help with trouble shooting

Our final poster can be found in the root directory or [here](https://github.com/STAT540-UBC/team_Methylhomies/blob/master/Poster.pdf)





