Allen Developing Human Brain 450K Data Normalization
========================================================
## By: Sumaiya Islam
## Date updated: August 10, 2016
  
### Script contents:
  - brain cell type correction of post-mortem human brain samples analyzed by Illumina HM450K platform from Allen Brain Institute (http://download.alleninstitute.org/brainspan/Methylation/). 
   
### A. Set up working directory & packages

R version 3.2.3 (2015-12-10)

We will initially set our working directory and load our libraries.

```{r}
install.packages("cets_0.99.2.tar.gz")
library(cets)
library(wateRmelon)
library(lumi)
```

##### We will use an open-source R package, Cell EpigenoType Specific (CETS) mapper, designed for the quantification and normalization of differing neuronal proportions in genome-scale DNA methylation datasets. The application of CETS quantification and transformation can reduce cell type heterogeneity and improve replicability of epigenetic findings in the brain across cohorts. It is important to correct for differences in cell type proportions in methylation data as these differences can lead to false discoveries in the methylation analysis (Guintivano et al 2013 *Epigenetics*).

Load calibration data set

The calibration dataset included in the cetsBrain package contains FACS-sorted neuron and glia samples from a cohort of depression patients and control subjects. The package contains two objects: A data matrix (brain) of sample methylation profiles for the top 10,000 most differentially methylated probes between neurons and glia and (pdBrain) a sample annotation data frame:

```{r}
# load "brain dataset" from data file in cetsBrain
load("~/Allen_Brain/cetsBrain.rda") # click on cetsBrain.rda file to place in workspace
dim(brain)
brain[1:3, 1:4]
head(pdBrain)
```

Load Allen methylation datasets post normalization

```{r}
load("Allen_brain_BMIQ.RData") # load normalized Allen 450K data
dim(betas_norm.fin)
```


Create the neuron and glia reference profiles:

```{r}
modelIdx <- list(neuron = pdBrain$celltype == "N", glia = pdBrain$celltype ==  "G")
 # getReference returns a 2-column matrix, representing reference profiles for the two cell types.
refProfile <- getReference(brain, modelIdx)
head(refProfile)
```

#### For the cortex dataset

Estimate the neuronal proportion:

The estProportion function returns an estimate of the percentage of cell type in the first column of its profile argument (neurons in this case). 
```{r}
prop <- estProportion(betas_norm.fin, profile = refProfile)
prop<-as.data.frame(prop)
prop$glia<-apply(prop,1,function(x) 1-x)
colnames(prop)<- c("neuron", "glia")
head(prop)
write.csv(prop, file = "Allen_CETS_Neuronal_Glia_Proportions.csv", row.names=T)
summary(prop)
plot(density(prop$neuron)) 
```


We will now perform cell-type correction based on the neuronal/glial proportions 

```{r}
all(rownames(prop)%in%colnames(betas_norm.fin))
brain.cor.dat<- as.data.frame(betas_norm.fin)

# fit methylation data for each probe in the dataset by the neuronal proportion
avebeta.lm<-apply(brain.cor.dat, 1, function(x){
  brain.sub<-prop[colnames(brain.cor.dat),]
  lm(x~neuron,data=brain.sub)
})

# obtain residuals for each probe across all samples (as a matrix)
residuals<-t(sapply(avebeta.lm, function(x)residuals(summary(x))))
head(residuals)
colnames(residuals)<-colnames(brain.cor.dat)

# generate adjusted residuals by adding the mean beta of each probe to the residuals
adj.residuals<-residuals+matrix(apply(brain.cor.dat, 1, mean), nrow=nrow(residuals), ncol=ncol(residuals))


r1<-as.data.frame(adj.residuals)
head(brain.cor.dat)
# check difference between corrected and uncorrected methylation data
all.equal(r1,brain.cor.dat)
```

To make sure we do not induce any NAs into the dataset when we convert the beta values back M-values (by log2 transformation), we need to ensure we do not have any corrected beta values that are greater or equal to zero or any beta values that are greater than 1. 

```{r}
adj.residuals[adj.residuals<=0]<-0.001 # convert any values that are less than or equal to zero to 0.001
adj.residuals[adj.residuals>1]<-0.999 # convert any values that are greater than 1 to 0.999
adj.M.values<-beta2m(adj.residuals)
any(is.na(adj.M.values)) # should be FALSE indicating there are no NAs
```

Save corrected dataset: 

```{r}
Allen.cell.corrected<-adj.residuals
save(Allen.cell.corrected, file="Allen_Dev_Brain_450K_Corrected.RData")
```