Hannon et al. (2015) 450K Data Cell Type Prediction
========================================================
## Original author: Sumaiya Islam
## Updated by: Samantha Schaffner
## Date updated: March 20, 2017
  
### Script contents:
  - brain cell type correction of post-mortem human brain samples analyzed by Illumina HM450K platform from Jonathan Mill's research group (PMC4844197). 
   
### A. Set up working directory & packages

R version 3.2.3 (2015-12-10)

We will initially set our working directory and load our libraries.

```{r}
install.packages("cets_0.99.2.tar.gz")
library(cets)
library(wateRmelon)
library(lumi)
library(dplyr)
```

##### We will use an open-source R package, Cell EpigenoType Specific (CETS) mapper, designed for the quantification and normalization of differing neuronal proportions in genome-scale DNA methylation datasets. The application of CETS quantification and transformation can reduce cell type heterogeneity and improve replicability of epigenetic findings in the brain across cohorts. It is important to correct for differences in cell type proportions in methylation data as these differences can lead to false discoveries in the methylation analysis (Guintivano et al 2013 *Epigenetics*).

Load calibration data set

The calibration dataset included in the cetsBrain package contains FACS-sorted neuron and glia samples from a cohort of depression patients and control subjects. The package contains two objects: A data matrix (brain) of sample methylation profiles for the top 10,000 most differentially methylated probes between neurons and glia and (pdBrain) a sample annotation data frame:

```{r}
# load "brain dataset" from data file in cetsBrain
load("~/team_Methylhomies/cetsBrain.rda") # click on cetsBrain.rda file to place in workspace
dim(brain)
brain[1:3, 1:4]
head(pdBrain)
```

Load methylation datasets post normalization

```{r}
load("GSE43414_filtered.RData") # load normalized 450K data
dim(GSE43414_filtered)
```


Create the neuron and glia reference profiles:

```{r}
modelIdx <- list(neuron = pdBrain$celltype == "N", glia = pdBrain$celltype ==  "G")
 # getReference returns a 2-column matrix, representing reference profiles for the two cell types.
refProfile <- getReference(brain, modelIdx)
head(refProfile)
```

#### For the brain datasets

Estimate the neuronal proportion:

The estProportion function returns an estimate of the percentage of cell type in the first column of its profile argument (neurons in this case). 
```{r}
prop <- estProportion(GSE43414_filtered, profile = refProfile)
prop<-as.data.frame(prop)
prop$glia<-apply(prop,1,function(x) 1-x)
colnames(prop)<- c("neuron", "glia")
head(prop)
write.csv(prop, file = "cellprop_uncor.csv", row.names=T)
summary(prop)
plot(density(prop$neuron), main="Neuronal Proportion Density") 
```
