Hannon et al. (2017) 450K Data Probe Filtering
========================================================
## Author: Hilary Brewis (adapted from Kobor lab scripts, original authors credited)
## Edited and compiled by: Samantha Schaffner
## Date updated: March 18, 2017
  
### Script contents:
  - normalization of post-mortem human brain samples analyzed by Illumina HM450K platform Jonathan Mill's research group (PMC4844197). 
  
Data Acquisition and Formatting
=======================================

### Original author: Rachel Edgar
### Edited by: Hilary Brewis and Samantha Schaffner
### Date updated: March 16, 2017

**Background**

Illumina Infinium HumanMethylation450 BeadChip supersedes its predecessors because of its increased coverage of the genome with a more diverse set of genomic categories. However, to get this improved coverage two different probe designs are used. These two types of probes have been shown to exhibit widely different methylation distributions causing a potential bias in the averaged methylation values [(Dedeurwaerder et al 2011)](https://www.ncbi.nlm.nih.gov/pubmed/22126295/). 
[Morris and Beck (2015)](https://www.ncbi.nlm.nih.gov/pubmed/25233806) have reviewed the commonly bioinformatic methods used to correct for this bias.  

**Acquiring Beta data**

The raw signal intensities for this study can be downloaded from [GEO (GSE43414)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE43414).

**Formatting Beta data**
```{r format beta values}

GSE43414 <-read.csv("/home/sschaffner/team_Methylhomies/GSE43414_betaqn_geo_all_cohorts.csv")
load("Brain_meta_matched_GSE43414.RData")

barcodes <- Brain_matched$barcode
GSE43414_brain <- as.data.frame(GSE43414$X)
names <- list()
for (i in 2:ncol(GSE43414)){
  for (j in 1:length(barcodes)){
    if (paste(substr(colnames(GSE43414)[i], start=2, stop=18)) == barcodes[j]){
      GSE43414_brain <- cbind(GSE43414_brain,GSE43414[,i])
      names <- c(names, barcodes[j])
}}}
colnames(GSE43414_brain) <- c("X", names)

rownames(GSE43414_brain) <- GSE43414_brain$X
GSE43414_brain <- GSE43414_brain[,-1]
save(GSE43414_brain, file="GSE43414_brain.RData")

Brain_matched<-Brain_matched[which((Brain_matched$barcode)%in%colnames(GSE43414_filtered)),]
save(Brain_matched, "Brain_meta_matched_GSE43414.RData")
```
   

Probe Filtering
======================

## Original author: Rachel Edgar
## Edited by: Hilary Brewis
## Updated: March 15, 2017

### Data set up and probe filter

I first loaded the files from the meta data pre-processing steps and formatted it in prepration for probe filtering:
```{r load }
load("GSE43414_brain.RData")
load("Brain_meta_matched_GSE43414.RData")
```

```{r format meta matched}
Meta_matched<-Brain_matched[which(Brain_matched$barcode%in%colnames(GSE43414_brain)),]
Meta_matched<-Brain_matched[match(colnames(GSE43414_brain), Brain_matched$barcode),]
```

First I wanted to know how many probes are in the data set.
```{r CpGs}
CpGs<- rownames(GSE43414_brain)
length(CpGs)
```
Prior to filtering, there are **485577** probes.

I used two different annotation sets that were designed for probes used in Illumina HumanMethylation450 BeadChip:
(1) Platform GPL13534 (standard platform for the 450 BeadChip) which can download from [here](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL13534)  
(2) Platform GPL16304 (Price Platform, additional annotations to go on top of the standard GPL13534) which can download from [here](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL16304)

I opened both of these platforms in R:
```{r annotation platforms}
GPL13534<- read.delim("GPL13534.txt", comment.char="#")
Price_Platform <- read.delim("Price_Platform.txt", comment.char = "#")
```

I then merged the two annotation files:
```{r merge annotations}
annotation <- merge(Price_Platform, GPL13534, by ="ID")
```
From here I was ready to start filtering out probes. 

### Probe filtering

I started by removing SNP probes.
```{r SNp probes}
SNP_Probes<-CpGs[grep("rs", CpGs)]
GSE43414_filtered<-GSE43414_brain[!CpGs%in%SNP_Probes,] #SNP probes start with rs instead of cg  
dim(GSE43414_filtered) 
```
A total of **65** probes were removed, leaving me with **485512**.

I then wanted to remove probes that have a SNP within the probe:
```{r probes with SNPs}
snp_index <-annotation$ID[which(is.na(annotation$n_target.CpG.SNP))]
GSE43414_filtered<- GSE43414_filtered[which(rownames(GSE43414_filtered)%in%snp_index), ]
dim(GSE43414_filtered)
```
This time **20869** probes were removed, leaving me with **464643** probes

Probes that targeted the X and Y chromosome were then removed
```{r X and Y probes}
notXY <- annotation$ID[!annotation$CHR%in%c("X", "Y")]
GSE43414_filtered<- GSE43414_filtered[which(rownames(GSE43414_filtered)%in%notXY), ]
dim(GSE43414_filtered) 
```
**11475** probes were removed, leaving a total of **453168**

I also wanted to remove probes that had multiple annealing sites on the genome. First I removed probes that cross hybridized with either the X or Y chromosome.
```{r cross hybridizing probes}
xy_hit_index <- annotation$ID[which(annotation$XY_Hits == "XY_NO")]
GSE43414_filtered<- GSE43414_filtered[which(rownames(GSE43414_filtered)%in%xy_hit_index), ]
dim(GSE43414_filtered)
```
**10673** probes were removed, leaving a total of **442495**

I then removed probes that cross hybridized on any autosomes.
```{r cross hybridizing probes 2}
auto_hit_index <- annotation$ID[which(annotation$Autosomal_Hits == "A_NO")]
GSE43414_filtered<- GSE43414_filtered[which(rownames(GSE43414_filtered)%in%auto_hit_index), ]
dim(GSE43414_filtered)
```
**27415** probes were removed, leaving a total of **415080**

Finally I want to eliminate any of the probes with NAs:
```{r NA filter}
na_count_probe <-sapply(1:nrow(GSE43414_filtered), function(y) length(which(is.na(GSE43414_filtered[y,]))))
na_count_probe_good<-which(na_count_probe<(ncol(GSE43414_filtered)*0.05))
GSE43414_filtered<-GSE43414_filtered[na_count_probe_good,]
dim(GSE43414_filtered)
```
**176** probes were removed, leaving a total of **414904**

This filtered data set was finally saved as an Rdata file.
```{r save rdata file}
save(GSE43414_filtered, file="GSE43414_filtered.RData")
```
